### DISCUSSION TITLE
@pixi/Layers is completely broken

### DISCUSSION THREAD

**Discussion by edgib102** [pixijs/pixijs#8618] [Q&A] [answered]
So I've been trying for a few weeks now to get proper sorting of objects in my scene working. I tried the obvious sprite1.zindex = 2 method, nothing. 
So I tried to see if the layers module would work. Nope. The only thing that happens is what's drawn last is the one on top.

Would appreciate any help, because I feel like I'm going insane.

Here's some mockup code I wrote, any clues?
[Code Result](https://imgur.com/a/h9kU8fa)

```
import * as PIXI from 'pixi.js'
import { Layer } from '@pixi/layers'

let app = new PIXI.Application({
    width:window.innerWidth, 
    height:window.innerHeight,
    backgroundColor:0xb5c69a,
    view: document.getElementById('main-canvas')
});

let layer = new Layer();


const loader = PIXI.Loader.shared;

loader.baseUrl = "images"
loader.add(["untitled.png", "voxel.png"]);

loader.load();

loader.onComplete.add(() => {
    const sprite1 = new PIXI.Sprite.from(loader.resources['untitled.png'].texture);

    sprite1.width = sprite1.width *0.2;
    sprite1.height = sprite1.height *0.2;

    sprite1.parentLayer = layer;
    
    app.stage.addChild(sprite1);



    const sprite2 = new PIXI.Sprite.from(loader.resources['voxel.png'].texture);

    sprite2.width = sprite2.width *0.2;
    sprite2.height = sprite2.height *0.2;

    
    sprite2.parentLayer = layer;
 
    app.stage.addChild(sprite2);

    
    sprite1.zOrder = 2;
    sprite2.zOrder = 1;

    console.log(layer._sortedChildren);
});
```

**Comment by domis86** [comment]
```
    sprite1.zIndex = 2;
    sprite2.zOrder = 1;
```

shouldnt this be? :

```
    sprite1.zOrder = 2;
    sprite2.zOrder = 1;
```

**Reply by edgib102** [reply]
Ah, must be from when I was testing it. Still, makes no difference, the same behaviour applies. Will edit the post!

**Reply by bigtimebuddy** [reply]
Please make sure you do `PIXI.Sprite.from()` OR `new PIXI.Sprite()` but not `new PIXI.Sprite.from()`.

**Comment by domis86** [comment, marked as answer]
Ok - look at working example here: https://codesandbox.io/s/lt7ubu?file=/src/index.js

stage needs to be `Stage` from `"@pixi/layers"` package:
```
app.stage = new Stage();
```

you need to add layer to stage and enable sorting (see https://github.com/pixijs/layers/blob/master/src/Group.ts#L38-L39 ) :

```
layer.group.enableSort = true; 
app.stage.addChild(layer);
```

and:

```
  // To see elements in layer._sortedChildren this must be executed: app.stage.updateStage() .
  // It is executed during render (thanks to stage being Stage from '@pixi/layers' package)
  app.renderer.render(app.stage);
```


also as @bigtimebuddy  mentioned: use `PIXI.Sprite.from(` instead of `new PIXI.Sprite.from(`


Please also read: https://www.shukantpal.com/blog/pixijs/pixijs-layers-kit/

**Reply by edgib102** [reply]
Oh my god, I can't thank you enough. The headache is finally gone
